### PR TITLE
backend/feature : Notification Service Integration

### PR DESCRIPTION
**Pull request by khuzema786**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Notification Service Interface

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by RitikaHanish — APPROVED** [review]

File: lib/mobility-core/src/Kernel/External/Notification/Interface/PayTM.hs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    ToJSON a,
+    ToJSON b
+  ) =>
+  PayTM.PayTMConfig ->
+  Interface.NotificationReq a b ->
+  m ()
+notifyPerson config req = do
+  case req.auth.notificationToken of
+    Just notificationToken -> do
+      let templateName = ("PushNotification." <> show req.category <> (fromMaybe "" $ (("." <>) . show) <$> req.subCategory))
```
Comment (roman-bodavskiy) [inline lib/mobility-core/src/Kernel/External/Notification/Interface/PayTM.hs:24]: redundant brackets

File: lib/mobility-core/src/Kernel/External/Notification/Interface/PayTM.hs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                deviceType = [PayTM.ANDROIDAPP, PayTM.IOSAPP],
+                templateName = templateName,
+                dynamicParams = req.dynamicParams
+              }
+      PayTM.notifyPerson
+        config
+        notificationData
+    Nothing -> do
+      logTagInfo "PayTM" $ "notificationToken of a person " <> req.auth.recipientId <> " not found"
+      pure ()
```
Comment (roman-bodavskiy) [inline lib/mobility-core/src/Kernel/External/Notification/Interface/PayTM.hs:38]: redundant `pure ()`

File: lib/mobility-core/src/Kernel/External/Notification/PayTM/Client.hs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    MonadFlow m,
+    ToJSON a
+  ) =>
+  PayTM.PayTMConfig ->
+  PayTM.NotificationReq a ->
+  m ()
+notifyPerson config req = do
+  when (isNothing $ find (\el -> el == req.templateName) config.whitelistedTemplates) $ do
+    logTagInfo tag $ "message not sent as template " <> req.templateName <> " is not whitelisted"
+    pure ()
```
Comment (roman-bodavskiy) [inline lib/mobility-core/src/Kernel/External/Notification/PayTM/Client.hs:37]: redundant `pure ()`

File: lib/mobility-core/src/Kernel/External/Notification/PayTM/Client.hs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  ( CoreMetrics m,
+    EncFlow m r,
+    MonadFlow m,
+    ToJSON a
+  ) =>
+  PayTM.PayTMConfig ->
+  PayTM.NotificationReq a ->
+  m ()
+notifyPerson config req = do
+  when (isNothing $ find (\el -> el == req.templateName) config.whitelistedTemplates) $ do
```
Comment (roman-bodavskiy) [inline lib/mobility-core/src/Kernel/External/Notification/PayTM/Client.hs:35]: `unless (any (\el -> el == req.templateName) config.whitelistedTemplates)`

File: lib/mobility-core/src/Kernel/External/Notification/PayTM/Client.hs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    logTagInfo tag $ "message not sent as template " <> req.templateName <> " is not whitelisted"
+    pure ()
+  apiKey <- pure . PayTM.getPaytmApiKey =<< decrypt config.apiKey
+  clientId <- pure . PayTM.getPaytmClientId =<< decrypt config.clientId
```
Comment (roman-bodavskiy) [inline lib/mobility-core/src/Kernel/External/Notification/PayTM/Client.hs:39]: `clientId <- PayTM.getPaytmClientId <$> decrypt config.clientId`

By the way this little improvements could be found with hlint:
`hlint . --report`

**Review by piyushKumar-1 — APPROVED** [review]

**Pull request merged**